### ISSUE TITLE
Improve performance by making `LintTask` incremental

### ISSUE THREAD

**Issue by mateuszkwiecinski** [jeremymailen/kotlinter-gradle#264]
Looking at the [source code](https://github.com/jeremymailen/kotlinter-gradle/blob/9931d2965ef076100ea2a0bf1f42dfc66d320172/src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/LintTask.kt#L50), running the `lintKotlin` will result in all the sources being checked on each task re-run. 

To improve the performance, `LintTask` could run against changed files only ([documentation](https://docs.gradle.org/current/userguide/custom_tasks.html#incremental_tasks))

If this gets implemented, a similar thing could be adopted for `formatKotlin` task (with slightly bigger effort due to overlapping inputs and outputs), but would help implementing: https://github.com/jeremymailen/kotlinter-gradle/issues/155

**Comment by jeremymailen** [comment]
Yes, this one is on my wishlist. We're incremental at the sourceSet level, but if a sourceSet is large we might get a boost from only processing changed files in the sourceSet.
